### PR TITLE
gummy: init at 0.1

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Description of changes
This adds [gummy](https://github.com/Fushko/gummy), a brightness and display temperature manager for X11. It can automatically adjust based off of the contents on the screen which is pretty interesting. In the future I'll add a nixos/home-manager module for it as well, since you need to start up a daemon for it.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and “core” functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/gummy/default.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  passthru.tests.version = testVersion { package = gummy; };
+
+  meta = with lib; {
+    homepage = "https://github.com/Fushko/gummy";
+    description = "Brightness and temperature manager for X11";
+    longDescription = ''
+      CLI screen manager for X11 that allows automatic and manual brightness/temperature adjustments,
+      via backlight (currently only for embedded displays) and gamma. Multiple monitors are supported.
+    '';
+    mainProgram = "gummy";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/gummy/default.nix:63]:
```suggestion
```
that is the default

File: pkgs/tools/misc/gummy/default.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  # Fixes the "gummy start" command, without this it cannot find the binary.
+  # Setting this through cmake does not seem to work.
+  postPatch = ''
+    substituteInPlace src/gummy/gummy.cpp --replace \
+      "CMAKE_INSTALL_DAEMON_PATH" "\"${placeholder "out"}/libexec/gummyd\""
+  '';
+
+  preFixup = ''
+    substituteInPlace $out/lib/udev/99-gummy.rules --replace \
+      "/bin/chmod" "${coreutils}/bin/chmod"
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/gummy/default.nix:49]:
```suggestion
    substituteInPlace $out/lib/udev/99-gummy.rules \
      --replace "/bin/chmod" "${coreutils}/bin/chmod"
```

File: pkgs/tools/misc/gummy/default.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+  cmakeFlags = [
+    "-DUDEV_DIR=${placeholder "out"}/lib/udev"
+  ];
+
+  # Fixes the "gummy start" command, without this it cannot find the binary.
+  # Setting this through cmake does not seem to work.
+  postPatch = ''
+    substituteInPlace src/gummy/gummy.cpp --replace \
+      "CMAKE_INSTALL_DAEMON_PATH" "\"${placeholder "out"}/libexec/gummyd\""
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/gummy/default.nix:44]:
```suggestion
    substituteInPlace src/gummy/gummy.cpp \
      --replace "CMAKE_INSTALL_DAEMON_PATH" "\"${placeholder "out"}/libexec/gummyd\""
```
that is easier to append stuff and keep nice diffs.

**Comment by IvarWithoutBones** [comment]
Adressed all the suggestions :+1:

**Pull request merged**